### PR TITLE
Planet-Planet interaction

### PR DESCRIPTION
**Pull request by EV4g**
Added planet-planet interaction
Updated the attractMass function to also work between two planets, and not only a planet and a sun
Explode() can now also take a arraylist as input, which will be used in the future for planet collisions/destruction

### CODE REVIEW

**Comment by EV4g** [comment]
There are still things which need to be added, but planet-planet interactions work.
A repelling force should be added, so planets dont clutter and destroy the universe.

**Comment by EV4g** [comment]
Latest version is a stable version

File: Planet.pde [hunk 1]
```diff
@@ -10,14 +10,16 @@ class Planet {
   color col;
 
   boolean onScreen = true;
-
+  boolean explode_me;
```
Comment (TechnicJelle) [inline Planet.pde]: Please rename this variable to `explodeMe`, to comply with the naming style of the rest of the variables.

Please also move the ` = false` out of the constructor into the raw class

File: SolarSystem.pde [hunk 2]
```diff
@@ -125,20 +129,43 @@ void draw() {
   //Planet physics -->
   if (!mousePressed && simRunning && !simHalted) {
     for (int t = 0; t < TIMESTEPS_PER_FRAME * facSimSpeedMod; t++) {
-      for (int i = planets.size() - 1; i >= 0; i--) {
+      for (int i = planets.size() - 1; i >= 0; i--) {
```
Comment (TechnicJelle) [inline SolarSystem.pde]: Some trailing spaces

File: SolarSystem.pde [hunk 3]
```diff
@@ -275,7 +302,39 @@ PVector attractMass(Planet p) {
...
+  }
+  return false;
+}
+
+
+PVector attractMass(Planet p, Planet o) {
+  float d = dist(p.pos, o.pos);
+  float m = o.mass * p.mass;
+
+  if (d > p.radius + o.radius) {
```
Comment (TechnicJelle) [inline SolarSystem.pde]: Trailing space

File: SolarSystem.pde [hunk 4]
```diff
@@ -400,7 +470,7 @@ void keyPressed() {
       for (int i = planets.size() - 1; i >= 0; i--) {
         Planet p = planets.get(i);
         if (dist(mouseX, mouseY, p.pos.x, p.pos.y) < p.radius) {
-          explode(p, i);
+          explode(p, i, planets);
```
Comment (TechnicJelle) [inline SolarSystem.pde]: Feels redundant. `planets` is a global variable

File: Planet.pde [hunk 5]
```diff
@@ -28,7 +30,7 @@ class Planet {
     f.div(mass); //Do take the mass into account in F = m * a  ==>  a = F / m
     acc.add(f);
   }
-
+
```
Comment (TechnicJelle) [inline Planet.pde]: Trailing spaces

File: SolarSystem.pde [hunk 6]
```diff
@@ -125,20 +129,43 @@ void draw() {
...
           planets.remove(i);
 
-        //updated roche limit
-        if (dist(p.pos, sun) < 2.456 * SUN_RADIUS * pow((SUN_MASS * p.radius * p.radius * p.radius) / (p.mass * SUN_RADIUS * SUN_RADIUS * SUN_RADIUS), (1.0 / 3.0))) {
-          explode(p, i);
-        }       
+        //roche limit
+        if (dist(p.pos, sun) < 2.456 * SUN_RADIUS * pow((SUN_MASS * p.radius * p.radius * p.radius) / (p.mass * SUN_RADIUS * SUN_RADIUS * SUN_RADIUS), (1.0 / 3.0)) && enableRoche) {
+          explode(p, i, planets);
+        }
```
Comment (TechnicJelle) [inline SolarSystem.pde]: Trailing spaces

File: SolarSystem.pde [hunk 7]
```diff
@@ -125,20 +129,43 @@ void draw() {
...
+        if (enablePtPPhysics) {
+          for (int z = planets.size() - 1; z >= 0; z--) {
+            if (z != i) {
+              p.applyForce(attractMass(p, planets.get(z)));
+            }
+          }
+
+          //Planets explode upon impact
+          for (int z = planets.size() - 1; z >= 0; z--) {
+            if (z != i && dist(p.pos, planets.get(z).pos) < (p.radius + planets.get(z).radius) / 2 && degrees(PVector.angleBetween(planets.get(z).vel, p.vel)) > 10 && p.mass < 10 * planets.get(z).mass && p.radius < 10 * planets.get(z).radius) {
```
Comment (TechnicJelle) [inline SolarSystem.pde]: Trailing spaces

File: SolarSystem.pde [hunk 8]
```diff
@@ -275,7 +302,39 @@ PVector attractMass(Planet p) {
...
+    return new PVector(0, 0);
+  } else {
+    //float rsq = sq(sq(dist(o.pos, p.pos)));
+    //float q = m/rsq; 
+    float q = 0;
+    return PVector.sub(o.pos, p.pos).normalize().mult(FAC_GRAV * -q);
+  }
+}
+
+boolean roche(Planet p, Planet M) {
```
Comment (TechnicJelle) [inline SolarSystem.pde]: Trailing spaces

**Review by TechnicJelle — CHANGES_REQUESTED** [review]
Some style changes, but the rest is looking good!

**Comment by EV4g** [comment]
Changes have been made

**Review by TechnicJelle — APPROVED** [review]

**Pull request merged**